File: CS/CustomWizardExample/Wizard/Presenters/CustomChooseDataSourceTypePage.cs
using System.Collections.Generic;
using DevExpress.DataAccess.Sql;
using DevExpress.DataAccess.Wizard;
using DevExpress.DataAccess.Wizard.Model;
using DevExpress.DataAccess.Wizard.Presenters;
using DevExpress.DataAccess.Wizard.Views;
using DevExpress.Entity.ProjectModel;
using System;
using DevExpress.DataAccess.Wizard.Services;

namespace CustomWizardExample.Wizard.Presenters {
    public class CustomChooseDataSourceTypePage : ChooseDataSourceTypePage<DataSourceModel> {
        public CustomChooseDataSourceTypePage(IChooseDataSourceTypePageView view, IWizardRunnerContext context, IConnectionStorageService connectionStorageService, ISolutionTypesProvider solutionTypesProvider, SqlWizardOptions options) : base(view, context, connectionStorageService, solutionTypesProvider, options)
        {
        }

        public override Type GetNextPageType() {
            if (View.DataSourceType == CustomDataSourceType.XmlPersons)
                return typeof(ChoosePersonsPage);
            return base.GetNextPageType();
        }
    }
}

File: VB/CustomWizardExample/Wizard/Presenters/CustomChooseDataSourceTypePage.vb
Imports System.Collections.Generic
Imports DevExpress.DataAccess.Sql
Imports DevExpress.DataAccess.Wizard
Imports DevExpress.DataAccess.Wizard.Model
Imports DevExpress.DataAccess.Wizard.Presenters
Imports DevExpress.DataAccess.Wizard.Views
Imports DevExpress.Entity.ProjectModel
Imports System
Imports DevExpress.DataAccess.Wizard.Services

Namespace CustomWizardExample.Wizard.Presenters
    Public Class CustomChooseDataSourceTypePage
        Inherits ChooseDataSourceTypePage(Of DataSourceModel)

        Public Sub New(view As IChooseDataSourceTypePageView, context As IWizardRunnerContext, connectionStorageService As IConnectionStorageService, solutionTypesProvider As ISolutionTypesProvider, options As SqlWizardOptions)
            MyBase.New(view, context, connectionStorageService, solutionTypesProvider, options)
        End Sub

        Public Overrides Function GetNextPageType() As Type
            If View.DataSourceType Is CustomDataSourceType.XmlPersons Then
                Return GetType(ChoosePersonsPage)
            End If
            Return MyBase.GetNextPageType()
        End Function
    End Class
End Namespace